python/arcticdb/version_store/library.py

```diff
@@ -907,6 +907,35 @@ def read_metadata(self, symbol: str, as_of: Optional[AsOf] = None) -> VersionedI
         """
         return self._nvs.read_metadata(symbol, as_of)
 
+    def read_metadata_batch(self, symbols: List[str], as_ofs: Optional[List[AsOf]] = None) -> Dict[str, VersionedItem]:
```

Comment:
This should return a list not a dict to allow multiple AsOf queries on a single symbol. That would be more consistent with `read_batch`. We should also have some parameters objects (something like `MetadataReadRequest`) like in `read_batch` rather than this fiddly API where the i-th element of `as_ofs` corresponds to the i-th element of `symbols`.

Reply:
That's a very fair comment. I investigated on Friday and we currently are not in a position to return it in the C++ layer with minimum changes. Therefore, this PR will have to contain the necessary C++ changes to support it. We are working now on deciding the best solution in that layer.

Reply:
We want to:

1)  Use same signature that read batch. In the doctoring we should note that we will ignore all the params than read requests, other than asofs and symbol. Add warnings!
2) Return list honouring original order